homeassistant/config_entries.py

```diff
@@ -1614,9 +1614,11 @@ def _index_entry(self, entry: ConfigEntry) -> None:
             unique_id_hash = entry.unique_id
             if not isinstance(entry.unique_id, str):
                 # Guard against integrations using unhashable unique_id
-                # In HA Core 2024.9, we should remove the guard and instead fail
                 if not isinstance(entry.unique_id, Hashable):  # type: ignore[unreachable]
-                    unique_id_hash = str(entry.unique_id)
+                    raise TypeError(
```

Comment:
Because we raise here, we no longer log that happens below?

Reply:
You could set unique_id to a `int` or any other immutable type, which would be hashable. Since we allow unique_id to be `str` or `None`, this would log a message and not raise. This is what `test_hashable_non_string_unique_id` covers.

Reply:
🤦 I need to step away from my keyboard :D